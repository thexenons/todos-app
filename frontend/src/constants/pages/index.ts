import { lazy } from "react";

import { PageKey, Pages } from "../../types";

export const PAGES: Pages = {
	[PageKey.home]: {
		path: "/",
		title: "Home",
		component: lazy(() => import("../../pages/Home")),
	},
	[PageKey.login]: {
		path: "/login",
		component: lazy(() => import("../../pages/Login")),
	},
	[PageKey.register]: {
		path: "/register",
		component: lazy(() => import("../../pages/Register")),
	},

	[PageKey.test]: {
		path: "/test",
		title: "Test",
		loader: async () => {
			return await (await import("../../pages/Test")).getInitialData();
		},
		component: lazy(() => import("../../pages/Test")),
		isProtected: true,
		children: {
			[PageKey.todosListSingle]: {
				path: "/:id",
				title: "Todos List Single",
				loader: async (args) => {
					return await (
						await import("../../pages/Test/TodosListSingle")
					).getInitialData(args);
				},
				component: lazy(() => import("../../pages/Test/TodosListSingle")),
			},
		},
	},
	[PageKey.components]: {
		path: "/components",
		title: "Components",
		component: lazy(() => import("../../pages/Components")),
		hideLayout: true,
		children: {
			[PageKey.componentsButton]: {
				path: "/button",
				title: "Button",
				component: lazy(() => import("../../pages/Components/Button")),
			},
			[PageKey.componentsAcceptButton]: {
				path: "/accept-button",
				title: "AcceptButton",
				component: lazy(() => import("../../pages/Components/AcceptButton")),
			},
			[PageKey.componentsUpdateButton]: {
				path: "/update-button",
				title: "UpdateButton",
				component: lazy(() => import("../../pages/Components/UpdateButton")),
			},
			[PageKey.componentsCreateButton]: {
				path: "/create-button",
				title: "CreateButton",
				component: lazy(() => import("../../pages/Components/CreateButton")),
			},
			[PageKey.componentsDeleteButton]: {
				path: "/delete-button",
				title: "DeleteButton",
				component: lazy(() => import("../../pages/Components/DeleteButton")),
			},
			[PageKey.componentsCancelButton]: {
				path: "/cancel-button",
				title: "CancelButton",
				component: lazy(() => import("../../pages/Components/CancelButton")),
			},
			[PageKey.componentsPanel]: {
				path: "/panel",
				title: "Panel",
				component: lazy(() => import("../../pages/Components/Panel")),
			},
			[PageKey.componentsModal]: {
				path: "/modal",
				title: "Modal",
				component: lazy(() => import("../../pages/Components/Modal")),
			},
			[PageKey.componentsFloatingPanel]: {
				path: "/floating-panel",
				title: "FloatingPanel",
				component: lazy(() => import("../../pages/Components/FloatingPanel")),
			},
			[PageKey.componentsInputBase]: {
				path: "/input-base",
				title: "InputBase",
				component: lazy(() => import("../../pages/Components/InputBase")),
			},
		},
	},
};
