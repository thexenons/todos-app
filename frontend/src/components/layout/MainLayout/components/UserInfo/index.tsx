import type { FC } from "react";

import { PageKey } from "../../../../../constants";
import useUser from "../../../../../hooks/useUser";
import { logout } from "../../../../../state/user";
import Button from "../../../../atoms/Button";
import Link from "../../../../atoms/Link";

const UserInfo: FC = () => {
	const user = useUser();

	return (
		<div>
			{user ? (
				<div>
					<div>{user.username}</div>
					<Button onClick={logout}>Logout</Button>
				</div>
			) : (
				<div>
					<Link pageKey={PageKey.login}>Login</Link>
				</div>
			)}
		</div>
	);
};

export default UserInfo;
