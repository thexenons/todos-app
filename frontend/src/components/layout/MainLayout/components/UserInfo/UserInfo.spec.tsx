import { render } from "@testing-library/react";

import BaseTestComponent from "../../../../../tests/BaseTestComponent";
import { TEST_USER } from "../../../../../tests/constants";
import UserInfo from ".";

describe("UserInfo", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<UserInfo />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});

	it("Render with user", () => {
		localStorage.setItem("user", JSON.stringify(TEST_USER));

		const { container } = render(
			<BaseTestComponent>
				<UserInfo />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
