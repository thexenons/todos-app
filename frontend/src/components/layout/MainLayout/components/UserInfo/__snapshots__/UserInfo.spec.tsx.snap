// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserInfo > Render 1`] = `
<div>
  <div>
    <div>
      <a
        href="/login"
      >
        Login
      </a>
    </div>
  </div>
</div>
`;

exports[`UserInfo > Render with user 1`] = `
<div>
  <div>
    <div>
      <div>
        test username
      </div>
      <button
        class="styled__ButtonWrapper-sc-f8cd241f-0 bSHSQa"
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
